shell/platform/windows/window_top_level_message_handler.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_WINDOW_TOP_LEVEL_MESSAGE_HANDLER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_WINDOW_TOP_LEVEL_MESSAGE_HANDLER_H_
+
+#include <cstdint>
+
+#include <Windows.h>
+
+namespace flutter {
+
+class FlutterWindowsEngine;
+
+class WindowTopLevelMessageHandler {
+ public:
+  WindowTopLevelMessageHandler(FlutterWindowsEngine& engine);
```

Comment:
Could this accept a `PlatformHandler` instead of the engine? That would allow us to remove the `RequestApplicationQuit` from the engine

Reply:
Alternatively, I would consider making this class agnostic to Flutter. You could introduce a `WindowsLifecycleDelegate` with methods like `OnExitRequested`. The `FlutterWindowsEngine` could implement this lifecycle delegate and call the `PlatformHandler::RequestAppExit`.

Reply:
The former can be done so long as we expose the engine's platform handler, basically replacing `RequestApplicationQuit` with a getter for it. In the case of the latter, what is the advantage of making it agnostic to Flutter?